R/await.R

```diff
@@ -178,18 +179,19 @@ safe_call_once <- function(...) {
   tryCatch(call_once(...), error = function(e) e)
 }
 
-# .id is the first argument to f.
-call_once <- function(f, ..., .id = NULL, .status_key = "state",
+# .id is the first argument to f
+call_once <- function(f, ...,
+                      .status_key = "state",
                       .success_states = c("succeeded"),
                       .error_states = c("failed", "cancelled"), fname) {
-  response <- do.call(f, c(.id, list(...)))
+  response <- do.call(what = f, args = sapply(X = ..., FUN = list))
   status <- response[[.status_key]]
   if (is.null(status)) stop("Cannot find status")
 
   called <- any(status %in% .success_states)
 
   if (any(status %in% .error_states)) {
-    args <- c(.id, list(...))
+    args <- sapply(X = ..., FUN = list)
```

Comment:
calling `list(...)` will give you a list of the args. This is the right refactor though, git `.id.` out of there : )